src/SkeletalTrapezoidation.cpp

```diff
@@ -1861,24 +1859,37 @@ void SkeletalTrapezoidation::addToolpathSegment(const ExtrusionJunction& from, c
         generated_toolpaths.resize(inset_idx + 1);
     }
     assert((generated_toolpaths[inset_idx].empty() || !generated_toolpaths[inset_idx].back().junctions.empty()) && "empty extrusion lines should never have been generated");
-    if (!force_new_path
-        && !generated_toolpaths[inset_idx].empty()
-        && generated_toolpaths[inset_idx].back().is_odd == is_odd
-        && shorterThen(generated_toolpaths[inset_idx].back().junctions.back().p - to.p, 10)
-        && std::abs(generated_toolpaths[inset_idx].back().junctions.back().w - to.w) < 10
+    if (generated_toolpaths[inset_idx].empty()
+        || generated_toolpaths[inset_idx].back().is_odd != is_odd
+        || generated_toolpaths[inset_idx].back().junctions.back().perimeter_index != inset_idx // inset_idx should always be consistent
+        || ( ! is_odd && ! from_is_3way && ! to_is_3way && ( // if this is a normal even line ...
+            from.region_id != to.region_id // It should always have a consistent region_id
+            || generated_toolpaths[inset_idx].back().junctions.back().region_id != to.region_id
+            ))
```

Comment:
This condition is getting pretty long and complicated.

Might I suggest splitting parts of this condition in a variable(s). This reduces the complexity of the whole condition and documentation can be encoded in the variable name.

Reply:
Done